cogs/tmpc.py

```diff
@@ -0,0 +1,276 @@
+        self.need_init = True
+        if not first_init:
+            self.ainit.start()
+
+    @listener()
+    async def on_ready(self):
+        global first_init
+        if first_init:
+            first_init = False
+            self.ainit.start()
+
+    @loop()
+    async def ainit(self):
+        """
+        Loads the configuration for the module.
+        """
+        global moderator, bot_channels
+        # noinspection PyTypeChecker
+        async with AinitManager(bot=self.bot,
+                                loop=self.ainit,
+                                need_init=self.need_init,
+                                bot_channels=bot_channels,
+                                moderator=moderator) as need_init:
+            if need_init:
+                DiscordComponents(self.bot)
+
+    @group(pass_context=True,
+           name="tmpc")
+    async def tmpc(self, ctx: Context):
+        if not ctx.invoked_subcommand:
+            raise BadArgument
+        member: Union[Member, User] = ctx.author
+        logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
+
+    @tmpc.command(pass_context=True)
+    async def keep(self, ctx: Context):
+        """
+        Makes a Study Channel stay for a longer time.
+        """
+        document = await self.check_tmpc_channel(ctx)
+        if type(document) == GamingChannel:
+            raise MayNotUseCommandError
+        if document.deleteAt:
+            document.deleteAt = None
+            embed = Embed(title="Turned off keep",
+                          description=f"This channel will no longer stay after everyone leaves")
+        else:
+            key = ConfigurationNameEnum.DEFAULT_KEEP_TIME.value
+            time_difference: tuple[int, int] = (await self.config_db.find_one({key: {"$exists": True}}))[key]
+            document.deleteAt = datetime.now() + timedelta(hours=time_difference[0], minutes=time_difference[1])
+            embed = Embed(title="Turned on keep",
+                          description=f"This channel will stay after everyone leaves for "
+                                      f"{time_difference[0]} hours and {time_difference[1]} minutes")
+
+        await self.study_db.update_one({DBKeyWrapperEnum.CHAT.value: document.channel_id}, document.document)
+        await ctx.reply(embed=embed)
+        await TmpChannelUtil.check_delete_channel(document.voice, self.study_db, logger)
+
+    @tmpc.command(pass_context=True)
+    async def lock(self, ctx: Context):
+        """
+        Locks a Study or Gaming Channel.
+        """
+        document = await self.check_tmpc_channel(ctx)
+        guild: Guild = ctx.guild
+
+        key = ConfigurationNameEnum.STUDENTY.value
+        verified = guild.get_role(
+            (await PrimitiveMongoData(CollectionEnum.ROLES).find_one({key: {"$exists": True}}))[key])
+
+        await document.voice.set_permissions(verified, view_channel=False, connect=False)
```

Comment:
view_channel should be handled with its own command.

Reply:
fixed